### ISSUE TITLE
Dancing Lights

### ISSUE THREAD

**Issue by BAmercury** [templerobotics/electrical#1]
Lights that dance to music.

Play music from computer>serial into arduino. FFT and at different frequencies blink lights or change hue?

**Comment by BAmercury** [comment]
DUE: 2 Weeks from now

**Comment by spencer-melnick** [comment]
My current approach is to do an FFT for every 1ms of the audio, and average the magnitudes for different light intensities. Still need to add serial communications, audio playback, and improve light intensity calculations

**Comment by BAmercury** [comment]
@spencer-melnick Dude nice job! Got some great looking python code up. Let me know or @DanCostello98 if you need any help

**Comment by DanCostello98** [comment]
I'm available most days this week if you need any help just let me know @spencer-melnick

**Comment by BAmercury** [comment]
@DanCostello98 @spencer-melnick next time you guys stop by the lab, I'll show you the LED strip we will eventually need to interface too. It'll require some soldering and you might need to build a circuit? We'll see when I show you

**Comment by spencer-melnick** [comment]
I just realized that if either the rendering or FFT operations take too long the whole thing gets desynchronized, so I'm going to work on a fix for that. @DanCostello98 do you think you could write a function to send some 8-bit numbers over serial communication to an Arduino?

**Comment by DanCostello98** [comment]
@spencer-melnick yeah I can definitely do that. I'm kind of new to coding but I can probably figure it out

**Comment by spencer-melnick** [comment]
@DanCostello98 [here](https://playground.arduino.cc/Interfacing/Python)'s a quick overview of how to do serial communication with the Arduino and pySerial. Let me know if you need any help!

**Comment by DanCostello98** [comment]
@spencer-melnick sorry I haven't gotten the code to you yet, my laptop wasn't working for the past few days. It's fixed now so I'll finish up the code and get it to you asap
